### ISSUE TITLE
Returning 302 redirect as response

### ISSUE THREAD

**Issue by ubenzer** [grpc-ecosystem/grpc-gateway#607]
Hi, 

I am looking for a way to return 302 redirects to another URL as a response from grpc-gateway. In my research, I couldn't find a way to do it. In fact, I somewhat assume that it is not possible due to [this](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/errors.go#L15). 

It would be really nice to provide this functionality.

Please see this [SO](https://stackoverflow.com/questions/49878855/how-to-do-a-302-redirect-in-grpc-gateway/49884259) question of mine, that explains the same thing in detail.

**Comment by tmc** [comment]
I thought there was an issue around having exact control of the raw http response body but I'm not seeing it now. I agree this is a use case that should be straightforward.

**Comment by c4milo** [comment]
@ubenzer, you may be able to do that by using https://godoc.org/github.com/grpc-ecosystem/grpc-gateway/runtime#WithOutgoingHeaderMatcher. The question whether this is a good practice or not remains. Perhaps @tmc can help to shed some light since I'm dubious right now.

**Comment by mayank-dixit** [comment]
@ubenzer what did you end up doing? `WithOutgoingHeaderMatcher` worked or you ended up using some other approach?

**Comment by ubenzer** [comment]
I ended up implementing our logic on the client side. In one request I get the URL for the image and in the second request, we get the image itself by using the response of the first request.

This was implemented before I learned that such a thing `WithOutgoingHeaderMatcher` existed. I don't know if that would've been worked for me or not.

I'll close this issue as this is not required for me anymore.
